map.go

```diff
@@ -490,6 +490,16 @@ func handleMapCreateError(attr sys.MapCreateAttr, spec *MapSpec, err error) erro
 			return fmt.Errorf("map create: %w", haveFeatErr)
 		}
 	}
+	// BPF_MAP_TYPE_RINGBUF's max_entries must be a power-of-2 multiple of kernel's page size.
+	if errors.Is(err, unix.EINVAL) &&
+		(attr.MapType == sys.BPF_MAP_TYPE_RINGBUF || attr.MapType == sys.BPF_MAP_TYPE_USER_RINGBUF) {
+		pageSize := uint32(os.Getpagesize())
+		maxEntries := attr.MaxEntries
+		nPages := maxEntries / pageSize
+		if maxEntries%pageSize != 0 || !internal.IsPow(nPages) {
```

Comment:
@hengyoush I think kernel actually checks that `maxEntries` is a power of two, not `nPages`: https://elixir.bootlin.com/linux/v6.8.4/source/kernel/bpf/ringbuf.c#L194 Do you think that is correct?

Reply:
Thank you for the correction.